src/ZoneServer/World/Actors/Characters/Components/CollectionComponent.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Melia.Zone.World.Items;
+using Melia.Shared.Data.Database;
+using Melia.Shared.Scripting;
+using Melia.Zone.Network;
+using Melia.Zone.Scripting;
+using Yggdrasil.Logging;
+using Yggdrasil.Scheduling;
+using Yggdrasil.Util;
+
+namespace Melia.Zone.World.Actors.Characters.Components
+{
+	/// <summary>
+	/// A character's collection manager.
+	/// </summary>
+	/// <remarks>
+	/// An important thing to note about the collection system is that Collections are account-level.
+	/// </remarks>
+	public class CollectionComponent : CharacterComponent
+	{
+		private readonly object _syncLock = new object();
+		private readonly Dictionary<int, CollectionData> _collectionList = new Dictionary<int, CollectionData>();
+		private readonly Dictionary<int, List<int>> _collectionProgress = new Dictionary<int, List<int>>();		
+
+		/// <summary>
+		/// Creates new instance for character.
+		/// </summary>
+		/// <param name="character"></param>
+		public CollectionComponent(Character character)
+			: base(character)
+		{
+		}
+
+
+		/// <summary>
+		/// Adds a new empty Collection.
+		/// <param name="collectionId"></param>
+		public bool Add(int collectionId)
+		{
+			lock (_syncLock)
+			{
+				if (ZoneServer.Instance.Data.CollectionDb.TryFind(collectionId, out var collectionData) && !_collectionList.ContainsKey(collectionId))
+				{
+					_collectionList.Add(collectionId, collectionData);
+					_collectionProgress.Add(collectionId, new List<int>());
+					return true;
+				}
+				return false;
+			}
+		}
+
+
+		/// <summary>
+		/// Adds a Collection without informing the client.
+		/// </summary>
+		/// <remarks>
+		/// This is primarily used while the character and its quests are
+		/// loaded from the database.
+		/// </remarks>
+		/// <param name="collectionId"></param>
+		/// <param name="registeredItems"></param>
+		public void AddSilent(int collectionId, List<int> registeredItems)
+		{
+			lock (_syncLock)
+			{
+				if (ZoneServer.Instance.Data.CollectionDb.TryFind(collectionId, out var collectionData))
+				{
+					_collectionList.Add(collectionId, collectionData);
+					_collectionProgress.Add(collectionId, registeredItems);
+					if (registeredItems.Count == collectionData.RequiredItems.Count)
+					{
+						// TODO: Add the reward to the list of earned rewards
+					}
+				}
+			}
+		}
+
+		/// <summary>
+		/// Gets a collection and returns the list of items registered to it.
+		/// </summary>
+		/// <param name="collectionId"></param>
+		/// <param name="registeredItems"></param>
+		/// <returns></returns>
+		public bool TryGet(int collectionId, out List<int> registeredItems)
+		{
+			lock (_syncLock)
+				return _collectionProgress.TryGetValue(collectionId, out registeredItems);
+		}
+
+		/// <summary>
+		/// Returns the total number of registered collections
+		/// </summary>
+		/// <returns></returns>
+		public int Count()
```

Comment:
Simple methods like this that only return a "property" of the component should not be a full blown method, but rather a property.